packages/common/src/model/ModelTransformFunctions.ts

```diff
@@ -852,47 +755,148 @@ export async function transformModel(
       resources[localPath] = resources[uri]
       delete resources[uri]
     })
-    /*
-    const doUpload = async (uri, data) => {
-      const [savePath, fileName] = fileUploadPath(uri)
-      const args = {
-        path: savePath,
-        fileName,
-        body: data,
-        contentType: (await getContentType(uri)) || ''
-      }
-      return API.instance.service(fileBrowserPath).patch(null, args)
-    }
-    await Promise.all(Object.entries(resources).map(([uri, data]) => doUpload(uri, data)))
-    result = await doUpload(args.dst.replace(/\.glb$/, '.gltf'), Buffer.from(JSON.stringify(json)))
-    */
 
     await Promise.all(
       Object.entries(resources).map(async ([uri, data]) => {
-        const blob = new Blob([data], { type: fileTypeToMime(uri.split('.').pop()!)! })
-        await doUpload(blob, uri)
+        const blob = new Blob([data as BlobPart], { type: fileTypeToMime(uri.split('.').pop()!)! })
+        await doUpload(...toProjectAndFileName(uri, srcBaseURL), blob)
       })
     )
-    let finalPath = args.dst.replace(/\.[^.]*$/, '.gltf')
-    if (!finalPath.endsWith('.gltf')) {
-      finalPath += '.gltf'
+    await doUpload(
+      ...toProjectAndFileName(finalPath, srcBaseURL),
+      new Blob([JSON.stringify(json)], { type: 'application/json' })
+    )
+  }
+
+  finalPath = pathJoin(srcBaseURL, finalPath)
+  console.log(`Wrote ${modelFormat} file: ${finalPath}`)
+  return finalPath
+}
+
+export const transformModel = async (
+  srcURL: string,
+  modelOperations: ModelTransformParameters[],
+  onMetadata: (index: number, key: string, data: any) => void = (key, data) => {},
+  onProgress?: (progress: number, status: Status, numerator?: number, denominator?: number) => void
+): Promise<string[]> => {
+  onProgress?.(0, Status.TransformingModels)
+
+  const srcDocument = await (await loaderIO).read(srcURL)
+  const documents: Document[] = []
+  const textureOperations: TextureOperation[] = []
+  const numDocOperations = modelOperations.length
+
+  const textureUsages = new Map<string, Set<string>>()
+  {
+    const graph = srcDocument.getGraph()
+    for (const mat of srcDocument.getRoot().listMaterials()) {
+      const eeMat = mat.getExtension<EEMaterial>('EE_material')
+      const args = eeMat?.args
+      if (args == null) {
+        continue
+      }
```

Comment:
This code only works on EEMaterials. We need the heuristics to work on normal materials as well. You'll need to look at the structure of a typical gltf material in addition to our custom materials. Here are the fields of a typical file from our asset packs:
![image](https://github.com/user-attachments/assets/4f7f66cd-651f-422c-b422-ce5d24a9b58d)
You can easily infer usages of the base color maps and normal maps from the normal schema

Reply:
Just added support for scanning the child edges of Material, too. Stripping `Map` and `Texture` out of the graph edge names, to make it easier on the usage heuristic.